### PR TITLE
To Do List App Added Interactivity

### PR DESCRIPTION
**Pull request by mudasiri**
In this pull request i added interactivity to my To Do List App. Below are a highlight of the functionalities:

1. Added event Listeners to all check boxes 
2. Updated completed status when checkbox are clicked
3. Enabled clear all button to clear all completed tasks

### CODE REVIEW

**Review by andersonlebon — CHANGES_REQUESTED** [review]
### Hi @mudasiri 
# Required Changes ♻️

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- [ ] As the [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Store%20the%20updated%20array%20of%20items%20in%20local%20storage%2C%20so%20the%20user%20gets%20the%20correct%20list%20values%20after%20the%20page%20reloads.) is saying,you should store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

**Comment by mudasiri** [comment]
> 

I am not able to understand the meaning of what you are referencing to. If you could kindly elaborate @andersonlebon .

**Comment by mudasiri** [comment]
>     * [ ]  As the [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#:~:text=Store%20the%20updated%20array%20of%20items%20in%20local%20storage%2C%20so%20the%20user%20gets%20the%20correct%20list%20values%20after%20the%20page%20reloads.) is saying,you should store the updated array of items in local storage, so the user gets the correct list values after the page reloads.
> 
@andersonlebon  regarding this comment. My Code implemented it and it's working as desired according to requirement I do store and update the array before putting in local storage. Therefore I do not understand your review comment. Kindly give me a clear review comment. This is my second comment since you did not respond to the first one. Please respond so that i can move on with my projects.

File: src/helper.js [hunk 1]
```diff
@@ -76,6 +78,27 @@ export default class Helper {
...
+        this.toggleCompleted(id, currentStatus);
+      }));
+    }
+
+    static toggleCompleted = (id, currentStatus) => {
+      const taskList = Helper.getLocalStorageList();
+      taskList[id].completed = currentStatus;
+      Helper.setLocalStorageList(taskList);
+      Helper.showTaskItems();
+    }
```
Comment (mudasiri) [inline src/helper.js:100]: This part of my code, updates the array of changes in task status and stores in local storage @andersonlebon .

**Review by HAMMAS-SALEEM — APPROVED** [review]
# Status: Approved :green_circle: :1st_place_medal: :100:
success

## Hey @mudasiri :wave:

Great work on implementing the requested changes. Your project is complete. :blush:
There is nothing else to say about your project. :+1:

## Highlights :1st_place_medal:

:heavy_check_mark: Descriptive PR Title
:heavy_check_mark: Descriptive PR Summary
:heavy_check_mark: Following Gitflow
:heavy_check_mark: Following Requirements

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**